### ISSUE TITLE
My block size on my RockChip device is not 512 bytes, it's 2048 bytes

### ISSUE THREAD

**Issue by bvstraalen** [neo-technologies/rkflashtool#4]
rkflashtool: info: Flash Info:
	Manufacturer: Micron (4)
	Flash Size: 8192MB
	Block Size: 2048KB
	Page Size: 8KB
	ECC Bits: 40
	Access Time: 32
	Flash CS: <0> <1>

but I see blocksize being assumed to be 512 bytes in lots of places.  Is that going to work?

Everything in the device space is in terms of blocks

CMDLINE:androidboot.console=ttyFIQ0 init=/init initrd=0x62000000,0x00800000 mtdparts=rk29xxnand:0x00002000@0x00002000(misc),0x00008000@0x00004000(kernel),0x00008000@0x0000C000(boot),0x0001000@0x00014000(recovery),0x00001000@0x00015000(backup),0x00001000@0x00016000(cache),0x00001000@0x00017000(userdata),0x00002000@0x00018000(kpanic),0x00001000@0x0001A000(system),0x00001000@0x0001B000(user),-@0x0001C000(linux)

so, counting by blocks should work perfectly fine.  Hard to follow the code.

**Comment by julienchauveau** [comment]
I guess the size of 512 bytes is the logical block size while the 2048 Kb is the physical memory block size. These values are pretty common…

Do you have some issues using the software?

**Comment by bvstraalen** [comment]
The tool seems to work OK.   I do not *quite* know how to use it properly, but it does read my RK3188 device properly.

I’m just now trying to figure out how to use a Cloudsto firmware image file with the tool.  I don’t know if I have to rkunpack or not.  Just trying to think things through before pulling the trigger on things ;-)


Brian

> On Jan 22, 2016, at 5:46 PM, Julien Chauveau <notifications@github.com> wrote:
> 
> I guess the size of 512 bytes is the logical block size while the 2048 Kb is the physical memory block size. These values are pretty common…
> 
> Do you have some issues using the software?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/neo-technologies/rkflashtool/issues/4#issuecomment-174107267>.
> 

Brian Van Straalen         Lawrence Berkeley Lab
BVStraalen@lbl.gov         Computational Research
(510) 486-4976             Division (crd.lbl.gov)
